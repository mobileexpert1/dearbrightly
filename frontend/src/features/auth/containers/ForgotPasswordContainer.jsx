import React, { Component } from 'react';
import { connect } from 'react-redux';
import styled from 'react-emotion';
import { requestPasswordResetEmail } from '../actions/passwordResetActions';
import { ForgotPasswordForm } from '../components/ForgotPasswordForm';
import {
  getPasswordResetErrorMessage,
  isFetchingPasswordReset,
  isPasswordEmailSent,
} from 'src/features/auth/selectors/passwordResetSelectors';
import BottomNav from 'src/features/checkout/components/BottomNav';
import { getEnvValue } from 'src/common/helpers/getEnvValue';
import { GTMUtils } from 'src/common/helpers/gtmUtils';
const DEBUG = getEnvValue('DEBUG');

const Container = styled('div')`
    width: 100%;
    height: 100%;
    margin: 0;
    overflow-y: scroll;
    @media (min-height:568px) and (min-width: 576px) {
      display: flex;
      flex-direction: column;
      flex-wrap: wrap
    }
`
class ForgotPasswordContainer extends Component {
  state = {
    email: '',
    submitted: false,
  };

  componentDidMount() {

  }

  handleChange = ({ target }) => this.setState({ [target.name]: target.value, submitted: false });

  handleSubmit = () => {
    const { email } = this.state;

    this.setState({ submitted: true });

    if (email) {
      this.props.requestPasswordResetEmail(email);
    }
    GTMUtils.trackCall('forgot-password_reset_click');
  };

  componentDidUpdate(prevProps) {
    const { navigateCheckout, passwordEmailSent } = this.props;
    if (!prevProps.passwordEmailSent && passwordEmailSent && navigateCheckout) {
      navigateCheckout('sign up', false);
    }
  }

  render() {
    const { passwordResetErrorMessage, isFetching } = this.props;
    const { email, submitted } = this.state;

    const hasError = !!passwordResetErrorMessage || (submitted && !email);
    const errorMessage = passwordResetErrorMessage || 'Email is required';
    const hideBackButton = this.props.navigateBack ? false : true;

    return (
        <Container id={"container"}>
          <ForgotPasswordForm
              hasError={hasError}
              errorMessage={errorMessage}
              onChange={this.handleChange}
              value={email}
              isFetching={isFetching}
              topClass={this.props.topClass}
          />
          <BottomNav
              currentCheckoutStepName={"forgot password"}
              backButtonType={"arrow"}
              backButtonClick={this.props.navigateBack}
              backButtonTitle={"Back"}
              disableBackButton={hideBackButton}
              disableNextButton={false}
              hideNextButtonArrow={true}
              hideBackButton={hideBackButton}
              nextButtonClick={this.handleSubmit}
              nextTitle={"Reset Password"}
          />
        </Container>
    );
  }
}

const mapStateToProps = state => ({
  isFetching: isFetchingPasswordReset(state),
  passwordResetErrorMessage: getPasswordResetErrorMessage(state),
  passwordEmailSent: isPasswordEmailSent(state),
});

export default connect(
  mapStateToProps,
  { requestPasswordResetEmail },
)(ForgotPasswordContainer);