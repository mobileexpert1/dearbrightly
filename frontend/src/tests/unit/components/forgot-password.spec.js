import React from 'react';
import { mount } from 'enzyme';
import sinon from 'sinon';

import ForgotPasswordContainer from 'src/features/auth/containers/ForgotPasswordContainer';

const resetPasswordAction = () => {};

describe('ForgotPassword', () => {
    it('should render correctly', () => {
        const component = mount(
            <ForgotPasswordContainer resetPasswordAction={resetPasswordAction} />,
        );
        expect(component).toMatchSnapshot();
    });

    it('should handle change and submit', () => {
        const component = mount(
            <ForgotPasswordContainer resetPasswordAction={resetPasswordAction} />,
        );
        const instance = component.instance();
        const handleChange = sinon.spy(instance, 'handleChange');
        const handleSubmit = sinon.spy(instance, 'handleSubmit');
        expect(component.state('email')).toEqual('');
        expect(component.state('submitted')).toEqual(false);
        handleChange({ target: { name: 'email', value: 'user@email.com' } });
        handleSubmit({ preventDefault: () => {} });
        expect(component.state('submitted')).toEqual(true);
    });
});
