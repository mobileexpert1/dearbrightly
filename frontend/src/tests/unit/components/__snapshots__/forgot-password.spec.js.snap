// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotPasswordContainer should render correctly 1`] = `
<ForgotPasswordContainer
  resetPasswordAction={[Function]}
>
  <div
    className="forgot-password-container"
  >
    <div
      className="forgot-password-header"
    >
      Forgot Password
    </div>
    <div
      className="forgot-password-section"
    >
      <div
        className="forgot-password-text"
      >
        Fill in your email below to request a new password. An email will be sent to the address below containing a link to verify your email address.
      </div>
      <form
        className="forgot-password-form"
        name="form"
        onSubmit={[Function]}
      >
        <div
          className="forgot-password-form-section"
        >
          <div
            className="form-group"
          >
            <Input
              handleChange={[Function]}
              name="email"
              required={true}
              title="Email Address"
              type="text"
            >
              <div
                className="input-container"
              >
                <div
                  className="input-title"
                >
                  Email Address
                  <span
                    className="input-required"
                  >
                    *
                  </span>
                </div>
                <input
                  name="email"
                  onChange={[Function]}
                  type="text"
                />
              </div>
            </Input>
          </div>
        </div>
        <div
          className="forgot-password-form-section"
        >
          <Button
            onClick={[Function]}
            text="Reset Password"
            type="black"
          >
            <div
              className="black-button"
              onClick={[Function]}
            >
              Reset Password
            </div>
          </Button>
        </div>
      </form>
    </div>
  </div>
</ForgotPassword>
`;
